From 1f934482aea56dc5e21032b230bbbceebb5b290e Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 5 Jul 2020 11:06:23 +0300
Subject: [PATCH 54/54] mcm-imx8m-mini: env: Update config and main file

1) Set the correct SoM values in the fw_env.config.
2) Generate all available env device options by issuing:
fw_printenv -e

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 tools/env/fw_env.config | 33 +++------------------------------
 tools/env/fw_env_main.c |  7 ++-----
 2 files changed, 5 insertions(+), 35 deletions(-)

diff --git a/tools/env/fw_env.config b/tools/env/fw_env.config
index 053895a2c0..3222fd715c 100644
--- a/tools/env/fw_env.config
+++ b/tools/env/fw_env.config
@@ -6,33 +6,6 @@
 # be the same as the Environment size, which is valid for NOR and SPI-dataflash
 # Device offset must be prefixed with 0x to be parsed as a hexadecimal value.
 
-# NOR example
-# MTD device name	Device offset	Env. size	Flash sector size	Number of sectors
-/dev/mtd1		0x0000		0x4000		0x4000
-/dev/mtd2		0x0000		0x4000		0x4000
-
-# MTD SPI-dataflash example
-# MTD device name	Device offset	Env. size	Flash sector size	Number of sectors
-#/dev/mtd5		0x4200		0x4200
-#/dev/mtd6		0x4200		0x4200
-
-# NAND example
-#/dev/mtd0		0x4000		0x4000		0x20000			2
-
-# On a block device a negative offset is treated as a backwards offset from the
-# end of the device/partition, rather than a forwards offset from the start.
-
-# Block device example
-#/dev/mmcblk0		0xc0000		0x20000
-#/dev/mmcblk0		-0x20000	0x20000
-
-# VFAT example
-#/boot/uboot.env	0x0000          0x4000
-
-# UBI volume
-#/dev/ubi0_0		0x0		0x1f000		0x1f000
-#/dev/ubi0_1		0x0		0x1f000		0x1f000
-
-# UBI volume by name
-#/dev/ubi0:env		0x0		0x1f000		0x1f000
-#/dev/ubi0:env-redund	0x0		0x1f000		0x1f000
+# Block device for mcm-imx8m-mini
+/dev/mmcblk2boot0	0x1000	0x4000
+#/dev/mmcblk1		0x1000	0x4000
diff --git a/tools/env/fw_env_main.c b/tools/env/fw_env_main.c
index dd8cc8223e..91a20f2200 100644
--- a/tools/env/fw_env_main.c
+++ b/tools/env/fw_env_main.c
@@ -141,11 +141,8 @@ static void parse_common_args(int argc, char *argv[])
 			exit(EXIT_SUCCESS);
 			break;
 		case 'e':
-#ifdef CONFIG_SYS_MMC_ENV_PART
-			fprintf(stderr, "/dev/mmcblk%dboot%d 0x%x 0x%x\n", CONFIG_SYS_MMC_ENV_DEV, CONFIG_SYS_MMC_ENV_PART-1, CONFIG_ENV_OFFSET, CONFIG_ENV_SIZE);
-#else
-			fprintf(stderr, "/dev/mmcblk%d 0x%x 0x%x\n", CONFIG_SYS_MMC_ENV_DEV, CONFIG_ENV_OFFSET, CONFIG_ENV_SIZE);
-#endif
+			fprintf(stdout, "/dev/mmcblk2boot0 0x%x 0x%x\n", CONFIG_ENV_OFFSET, CONFIG_ENV_SIZE);
+			fprintf(stdout, "/dev/mmcblk1 0x%x 0x%x\n", CONFIG_ENV_OFFSET, CONFIG_ENV_SIZE);
 			exit(EXIT_SUCCESS);
 			break;
 		default:
-- 
2.11.0

